src/planner/constify_now.c

```diff
@@ -145,13 +145,34 @@ constify_now_expr(PlannerInfo *root, OpExpr *op)
 	else
 	{
 		OpExpr *op_inner = lsecond_node(OpExpr, op->args);
+		Const *const_offset = lsecond_node(Const, op_inner->args);
+		Assert(const_offset->consttype == INTERVALOID);
+		Interval *offset = DatumGetIntervalP(const_offset->constvalue);
 		/*
 		 * Sanity check that this is a supported expression. We should never
 		 * end here if it isn't since this is checked in is_valid_now_expr.
 		 */
 		Assert(linitial_node(FuncExpr, op_inner->args)->funcid == F_NOW);
 		linitial(op_inner->args) = make_now_const();
 
+		/*
+		 * If the interval has a day component then the calculation needs
+		 * to take into account daylight saving time switches and thereby a
+		 * day would not always be exactly 24 hours. We mitigate this by
+		 * adding a safety buffer to account for these dst switches when
+		 * dealing with intervals with day component. These calculations
+		 * will be repeated with exact values during execution.
+		 * Since dst switches seem to range between -1 and 2 hours we set
+		 * the safety buffer to 4 hours.
+		 */
+		if (offset->day != 0)
+		{
+			Const *now = linitial_node(Const, op_inner->args);
+			int64 now_value = DatumGetInt64(now->constvalue);
+			now_value -= 4 * USECS_PER_HOUR;
```

Comment:
Should this always be minus? 

I know it's a weird case, but some people do use times in the future, do we need to check whether we're adding or subtracting?

Reply:
yes should always be minus, we still do the actual interval calculation with this safety-adjusted value so whatever the rest of the expression says will be calculated with this value